monitoring/monitoring/internal/headertransport/headertransport.go

```diff
@@ -0,0 +1,28 @@
+package headertransport
+
+import "net/http"
+
+// AddHeaderTransport implements http.RoundTripper and allows us to inject
+// additional HTTP headers on requests
+type AddHeaderTransport struct {
+	T http.RoundTripper
+
+	additionalHeaders map[string]string
+}
+
+func (adt *AddHeaderTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	for header, value := range adt.additionalHeaders {
+		req.Header.Add(header, value)
+	}
+	return adt.T.RoundTrip(req)
+}
+
+func NewAddHeaderTransport(T http.RoundTripper, headers map[string]string) *AddHeaderTransport {
```

Comment:
Nit, but the current naming stutters (repeated "header transport") - consider a rename like:

```suggestion
func New(T http.RoundTripper, headers map[string]string) *Transport {
```

Then your code will say:

```
headertransport.New(...) // "new header transport"
```